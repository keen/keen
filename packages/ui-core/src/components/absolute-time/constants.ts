export const DATE_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
