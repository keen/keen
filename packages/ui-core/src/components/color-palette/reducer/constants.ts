export const SET_IS_DRAGGED = 'SET_IS_DRAGGED';
export const SET_COLORS = 'SET_COLORS';
export const SET_ACTIVE_COLOR = 'SET_ACTIVE_COLOR';
export const SET_ADD_COLOR_PICKER_OPEN = 'SET_ADD_COLOR_PICKER_OPEN';
export const SET_COLOR_PICKER_POSITION = 'SET_COLOR_PICKER_POSITION';
