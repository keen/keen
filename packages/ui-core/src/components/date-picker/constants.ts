export const DATE_FORMAT = 'YYYY-MM-DD';
export const TIME_FORMAT = 'HH:mm';
