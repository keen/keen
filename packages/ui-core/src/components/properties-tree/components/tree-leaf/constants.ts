export const SEPARATOR = '.';
export const PADDING = 15;
