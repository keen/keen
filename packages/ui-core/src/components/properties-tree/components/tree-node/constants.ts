export const PADDING = 15;
