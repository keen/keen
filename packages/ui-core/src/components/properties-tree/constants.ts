export const ELEMENT_HEIGHT = 36;
export const LIST_HEIGHT = 200;
