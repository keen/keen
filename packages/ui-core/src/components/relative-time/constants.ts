export const RELATIVITY_OPTIONS = [
  {
    label: 'this',
    value: 'this',
  },
  {
    label: 'previous',
    value: 'previous',
  },
];

export const THIS_RELATIVITY = 'this';
export const PREVIOUS_RELATIVITY = 'previous';
