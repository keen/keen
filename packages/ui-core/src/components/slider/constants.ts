export const SET_DIMENSION = 'SET_DIMENSION';
export const SET_VALUE = 'SET_VALUE';
export const SET_DRAG_POSITION = 'SET_DRAG_POSITION';
export const SET_DRAG_STATE = 'SET_DRAG_STATE';
export const SET_DRAG_MOVING = 'SET_DRAG_MOVING';

export const DRAG_CONTROL_ID = '@keen-drag-control';

export const TOOLTIP_CONTENT_MAX_WIDTH = 200;
