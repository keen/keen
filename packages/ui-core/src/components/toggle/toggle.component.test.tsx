import React, { ComponentProps } from 'react';
import { render as rtlRender, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import Toggle from './toggle.component';
import { KEYBOARD_KEYS } from '../../constants';

const render = (overProps: Partial<ComponentProps<typeof Toggle>> = {}) => {
  const props = {
    onChange: jest.fn(),
    ...overProps,
  };
  const wrapper = rtlRender(<Toggle {...props} />);

  return {
    wrapper,
    props,
  };
};

test('should render Toggle component', () => {
  const { wrapper } = render();
  expect(wrapper).toMatchSnapshot();
});

test('should call onChange if provided', () => {
  const {
    wrapper: { getByRole },
    props,
  } = render();
  const element = getByRole('switch');
  fireEvent.click(element);

  expect(props.onChange).toHaveBeenCalled();
});

test('should call onChange using keyboard', () => {
  const {
    wrapper: { getByRole },
    props,
  } = render();
  const element = getByRole('switch');
  fireEvent.keyDown(element, { key: 'Enter', keyCode: KEYBOARD_KEYS.ENTER });

  expect(props.onChange).toHaveBeenCalled();
});
