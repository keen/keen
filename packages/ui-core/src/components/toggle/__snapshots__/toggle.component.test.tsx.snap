// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Toggle component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 60px;
  height: 22px;
  position: relative;
  cursor: pointer;
}

.c0:focus-visible {
  outline: 1px solid #85B4C3;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  width: 60px;
  height: 22px;
  border-radius: 22px;
  overflow: hidden;
  background-color: rgba(225,226,228,0.7);
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 60px;
  height: 22px;
  border-radius: 22px;
  box-sizing: border-box;
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  background-color: #6C9A74;
}

.c3 {
  position: relative;
  width: 60px;
  height: 22px;
  padding: 4px 10px;
  border-radius: 22px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Lato Regular',sans-serif;
  font-size: 11px;
  text-transform: uppercase;
  box-sizing: border-box;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  height: 22px;
  width: 22px;
  border-radius: 50%;
  box-sizing: border-box;
  background-color: #FFFFFF;
  border: 2px solid #487650;
}

<body>
    <div>
      <div
        aria-checked="false"
        class="c0"
        data-testid="toggle"
        role="switch"
        tabindex="0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
            style="transform: scaleX(0) translateZ(0);"
          />
        </div>
        <div
          class="c3"
          style="justify-content: flex-end; color: rgb(79, 91, 95);"
        >
          off
        </div>
        <div
          class="c4"
          style="transform: translateX(-1px) translateZ(0);"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  width: 60px;
  height: 22px;
  position: relative;
  cursor: pointer;
}

.c0:focus-visible {
  outline: 1px solid #85B4C3;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  width: 60px;
  height: 22px;
  border-radius: 22px;
  overflow: hidden;
  background-color: rgba(225,226,228,0.7);
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 60px;
  height: 22px;
  border-radius: 22px;
  box-sizing: border-box;
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  background-color: #6C9A74;
}

.c3 {
  position: relative;
  width: 60px;
  height: 22px;
  padding: 4px 10px;
  border-radius: 22px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Lato Regular',sans-serif;
  font-size: 11px;
  text-transform: uppercase;
  box-sizing: border-box;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  height: 22px;
  width: 22px;
  border-radius: 50%;
  box-sizing: border-box;
  background-color: #FFFFFF;
  border: 2px solid #487650;
}

<div>
    <div
      aria-checked="false"
      class="c0"
      data-testid="toggle"
      role="switch"
      tabindex="0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
          style="transform: scaleX(0) translateZ(0);"
        />
      </div>
      <div
        class="c3"
        style="justify-content: flex-end; color: rgb(79, 91, 95);"
      >
        off
      </div>
      <div
        class="c4"
        style="transform: translateX(-1px) translateZ(0);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
